projects/iio_demo_freeRTOS/src/examples/iio_example/iio_example.c

```diff
@@ -0,0 +1,108 @@
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "iio_example.h"
+#include "iio_adc_demo.h"
+#include "iio_dac_demo.h"
+#include "common_data.h"
+#include "no_os_util.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/***************************************************************************//**
+ * @brief IIO example main execution.
+ *
+ * @return ret - Result of the example execution. If working correctly, will
+ *               execute continuously function iio_app_run and will not return.
+*******************************************************************************/
+int iio_example_main()
+{
+	int32_t status;
+
+	/* adc instance descriptor. */
+	struct adc_demo_desc *adc_desc;
+
+	/* dac instance descriptor. */
+	struct dac_demo_desc *dac_desc;
+
+	/* IIO application descriptor. */
+	struct iio_app_desc *app;
+
+	/* IIO application initialization parameters. */
+	struct iio_app_init_param app_init_param = { 0 };
+
+	struct iio_data_buffer adc_buff = {
+		.buff = (void *)ADC_DDR_BASEADDR,
+		.size = MAX_SIZE_BASE_ADDR
+	};
+
+	struct iio_data_buffer dac_buff = {
+		.buff = (void *)DAC_DDR_BASEADDR,
+		.size = MAX_SIZE_BASE_ADDR
+	};
+
+	status = adc_demo_init(&adc_desc, &adc_init_par);
+	if (status)
+		return status;
+
+	status = dac_demo_init(&dac_desc, &dac_init_par);
+	if (status)
+		return status;
+
+	struct iio_app_device devices[] = {
+		IIO_APP_DEVICE("adc_demo", adc_desc,
+			       &adc_demo_iio_descriptor,&adc_buff, NULL, NULL),
+		IIO_APP_DEVICE("dac_demo", dac_desc,
+			       &dac_demo_iio_descriptor,NULL, &dac_buff, NULL)
+	};
+
+	app_init_param.devices = devices;
+	app_init_param.nb_devices = NO_OS_ARRAY_SIZE(devices);
+	app_init_param.uart_init_params = iio_demo_uart_ip;
+
+	status = iio_app_init(&app, app_init_param);
+	if (status)
+		return status;
+
+	return iio_app_run(app);
```

Comment:
this is something that is missing from other projects.. but in case of an error adc_demo_remove, dac_demo_remove and iio_app_remove should be called before returning to avoid memory leaks

Reply:
added